### PR TITLE
Measure app binary size changes

### PR DESCRIPTION
**Pull request by Mastersam07**
<h1>What does this PR accomplish?</h1>
Build app on every push and pull request made to master branch. Compare the app on the master branch with the new changes coming from the push/pull request, analyzing both apps and listing the change in binary size.

<h2>Did you add any dependencies?</h2>
diffuse artifacts - github actions

---
Issue #1013 

---

#### Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).

- [x] REQUIRED: Do you have an Issue **assigned to you** for this PR?
- [x] Provided detailed pull request description and a succinct title
- [ ] Tested your changes, especially after any code review iterations.
- [ ] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [ ] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [content/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/content/credits.yaml) file (if you want it to be).

### CODE REVIEW

File: .github/workflows/compare-apk.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    needs: [build-master, build-new]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Download App bundles
+        uses: actions/download-artifact@v1
+        with:
+          name: apk
+      - name: Change diffuse permissions
+        run: chmod +x ./diffuse
```
Comment (epicfaace) [inline .github/workflows/compare-apk.yml:69]: Is it possible to download diffuse from a URL instead of checking in the binary file to this repo?

Comment (advayDev1) [reply .github/workflows/compare-apk.yml:69]: Indeed, can’t checkin the binary unless you’ve met all licensing requirements of the full binary.

File: .github/workflows/compare-apk.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+name: Compare APK
+on:
+  push:
```
Comment (advayDev1) [inline .github/workflows/compare-apk.yml:3]: Does this need to run on push?

File: .github/workflows/compare-apk.yml (hunk 1 above)
Comment (Mastersam07) [reply .github/workflows/compare-apk.yml:69]: We could clone the GitHub repository for diffuse, generate the diffuse executable and add to path with the GitHub actions.

File: .github/workflows/compare-apk.yml (hunk 2 above)
Comment (Mastersam07) [reply .github/workflows/compare-apk.yml:3]: If that's not necessary, we could perform the action on pull request.

**Comment by Mastersam07** [comment]
@advayDev1 Set the action to be performed on pull request

**Comment by Mastersam07** [comment]
@epicfaace removed the diffuse binary file, then set up via GitHub actions

**Comment by advayDev1** [comment]
I think we should make a few structural changes:
* Make the android app bundle from `client-build.yaml` into an artifact
* Move your actions #1, 3, and 4 into a single action that runs after android build in `client-build.yaml`
* Remove the action to upload the diffuse binary.  Instead just use the binary in the same action.


Its important that we conserve our build minutes.

**Comment by Mastersam07** [comment]
> I think we should make a few structural changes:
> 
> * Make the android app bundle from `client-build.yaml` into an artifact
> * Move your actions #1, 3, and 4 into a single action that runs after android build in `client-build.yaml`
> * Remove the action to upload the diffuse binary.  Instead just use the binary in the same action.
> 
> Its important that we conserve our build minutes.

Analysis and comparison of the appbundle do not give much information compared to apk. I will have to build the apk instead of appbundle then make it an artifact.

**Comment by advayDev1** [comment]
@Mastersam07 can you explain what you mean by the appbundle isn't as useful?

**Comment by Mastersam07** [comment]
> @Mastersam07 can you explain what you mean by the appbundle isn't as useful?

Running the diffuse command and comparing APKs gives us the changes in app binary size - this includes changes in assets, methods, strings, resources, types, fields, native code(information you get when analyzing APKs via android studio apk analyzer). We won't be able to get these information with an appbundle.

**Comment by advayDev1** [comment]
I see.  I want to hear @hspinks thoughts on this.  I think this is a lot of build minutes to spend on comparing file sizes.

**Pull request closed without merging**